### PR TITLE
Minor update to improve compilation with MSVC

### PR DESCRIPTION
**Pull request by TLCFEM**

### CODE REVIEW

File: src/precon/lis_precon_saamg.c [hunk 1]
```diff
@@ -314,8 +314,8 @@ LIS_INT lis_precon_psd_update_saamg(LIS_SOLVER solver, LIS_PRECON precon)
 	#endif
 
 	LIS_DEBUG_FUNC_OUT;
-    return LIS_SUCCESS;
```
Comment (TLCFEM) [inline src/precon/lis_precon_saamg.c:317]: To suppress no return, `-Werror=return-type`.

File: src/system/lis_time.c [hunk 2]
```diff
@@ -41,7 +41,7 @@
 #include <stdarg.h>
 #include <time.h>
 
-#ifdef WIN32
```
Comment (TLCFEM) [inline src/system/lis_time.c:44]: MSVC defines _WIN32 instead.

**Comment by anishida** [comment]
Hi,

Thank you for your comments.
Could you provide us with some more information?
Regarding AMG preconditioning, do you mean that 
You get a warning messssage?
For WIN32, does it mean that it should be _WIN32 instead?
For ishift, you are right. We will correct them.

Sincerely,
Akira Nishida

From: Theodore ***@***.*** 
Sent: Wednesday, June 14, 2023 12:41 AM
To: anishida/lis ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [anishida/lis] Minor update to improve compilation with MSVC (PR #32)

________________________________________
You can view, comment on, or merge this pull request online at:
  https://github.com/anishida/lis/pull/32
Commit Summary
• de552e4 Minor update to improve compilation with MSVC
File Changes 
(2 files)
• M src/precon/lis_precon_saamg.c (2) 
• M src/system/lis_time.c (2) 
Patch Links:
• https://github.com/anishida/lis/pull/32.patch
• https://github.com/anishida/lis/pull/32.diff

**Comment by TLCFEM** [comment]
> Hi, Thank you for your comments. Could you provide us with some more information? Regarding AMG preconditioning, do you mean that You get a warning messssage? For WIN32, does it mean that it should be _WIN32 instead? For ishift, you are right. We will correct them. Sincerely, Akira Nishida From: Theodore ***@***.*** Sent: Wednesday, June 14, 2023 12:41 AM To: anishida/lis ***@***.***> Cc: Subscribed ***@***.***> Subject: [anishida/lis] Minor update to improve compilation with MSVC (PR #32)
> […](#)
> ________________________________________ You can view, comment on, or merge this pull request online at: #32 Commit Summary • [de552e4](https://github.com/anishida/lis/commit/de552e41b8cec5168b2bf4f919c24af574700d96) Minor update to improve compilation with MSVC File Changes (2 files) • M src/precon/lis_precon_saamg.c (2) • M src/system/lis_time.c (2) Patch Links: • https://github.com/anishida/lis/pull/32.patch • https://github.com/anishida/lis/pull/32.diff — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

Hi Akira,

For saamg, that function does not have a return statement if saamg is not enabled, moving the final return statement out of the macro if block suppresses the no return value warning.

For MSVC, it defines `_WIN32`, so `WIN32` does not detect MSVC, it thus inclues `<sys/time.h>` which is not available on Windows. See further: https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

**Comment by anishida** [comment]
Thank you for your reply.
So, you mean we should move "return LIS_SUCCESS;" out of the scope of the macro, and change WIN32 to _WIN32?

Sincerely,
Akira Nishida

**Comment by TLCFEM** [comment]
> we should move "return LIS_SUCCESS;" out of the scope of the macro

Yes, that is what is changed in this PR.

> change WIN32 to _WIN32?

It think gcc defines WIN32 on windows, so what I changed is from

```
#ifdef WIN32
```

to

```
#if defined(WIN32) || defined(_WIN32)
```

so that both would be covered.

**Comment by anishida** [comment]
Dear Theodore,

Thank you for your message.
We have updated the release based on your comment.

Sincerely,
Akira Nishida

**Comment by TLCFEM** [comment]
Thank you for the amazing library!

**Pull request closed without merging**